src/datta_lab_to_nwb/markowitz_gillis_nature_2023/basevideointerface.py

```diff
@@ -0,0 +1,82 @@
+"""Base class for converting raw video data."""
+from pynwb import NWBFile
+from datetime import datetime
+from pytz import timezone
+import h5py
+import numpy as np
+import pandas as pd
+from neuroconv.basedatainterface import BaseDataInterface
+from neuroconv.datainterfaces import VideoInterface
+
+
+class BaseVideoInterface(BaseDataInterface):
+    """Base video interface for markowitz_gillis_nature_2023 conversion"""
+
+    def __init__(self, data_path: str, metadata_path: str, timestamp_path: str):
+        super().__init__(
+            data_path=data_path,
+            metadata_path=metadata_path,
+            timestamp_path=timestamp_path,
+        )
+
+    def get_metadata(self) -> dict:
+        metadata = super().get_metadata()
+        with h5py.File(self.source_data["metadata_path"]) as file:
+            session_id = np.array(file["metadata"]["uuid"], dtype="U").item()
+            subject_id = np.array(file["metadata"]["acquisition"]["SubjectName"], dtype="U").item()
+            session_start_time = np.array(file["metadata"]["acquisition"]["StartTime"], dtype="U").item()
+            session_name = np.array(file["metadata"]["acquisition"]["SessionName"], dtype="U").item()
+        metadata["NWBFile"]["session_id"] = session_id
+        metadata["NWBFile"]["identifier"] = session_id
+        metadata["NWBFile"]["session_start_time"] = datetime.fromisoformat(session_start_time).astimezone(
+            timezone("US/Eastern")
+        )
+        metadata["NWBFile"]["session_description"] = session_name
+        metadata["Subject"]["subject_id"] = subject_id
+        metadata["Subject"]["sex"] = "U"  # TODO: Add dict of sexes from email
+
+        return metadata
+
+    def get_metadata_schema(self) -> dict:
+        metadata_schema = super().get_metadata_schema()
+        metadata_schema["properties"]["Behavior"] = {
+            "type": "object",
+            "properties": {
+                "CompassDirection": {
+                    "type": "object",
+                    "properties": {
+                        "reference_frame": {"type": "string"},
+                    },
+                },
+                "Position": {
+                    "type": "object",
+                    "properties": {
+                        "reference_frame": {"type": "string"},
+                    },
+                },
+            },
+        }
+        metadata_schema["properties"]["BehavioralSyllable"] = {
+            "type": "object",
+            "properties": {
+                "sorted_pseudoindex2name": {"type": "object"},
+                "id2sorted_index": {"type": "object"},
+                "sorted_index2id": {"type": "object"},
+            },
+        }
```

Comment:
@pauladkisson Though I did just now notice this block of metadata schema that seems to be from the MoseqInterface - should it belong here too? (Doesn't seem to be used pertaining to the videos)

Reply:
Yeah it's due to the way that I'm running `convert_session` here vs the other branches (`convert_moseq`, `convert_raw`, etc.).  I'll need to make an integration PR after these get merged.